src/devices/machine/idehd.cpp

```diff
@@ -379,22 +379,22 @@ void ata_mass_storage_device::security_error()
 
 attotime ata_mass_storage_device::seek_time()
 {
-	int sectors_per_cylinder =  m_num_heads * m_num_sectors;
+	uint32_t sectors_per_cylinder =  m_num_heads * m_num_sectors;
 
 	if (sectors_per_cylinder == 0 || m_num_cylinders == 0)
 		return attotime::zero;
 
-	int new_lba = lba_address();
-	int old_cylinder = m_cur_lba / sectors_per_cylinder;
-	int new_cylinder = new_lba / sectors_per_cylinder;
-	int diff = abs(old_cylinder - new_cylinder);
+	uint32_t new_lba = lba_address();
+	int16_t old_cylinder = m_cur_lba / sectors_per_cylinder;
+	int16_t new_cylinder = new_lba / sectors_per_cylinder;
+	uint16_t diff = abs(old_cylinder - new_cylinder);
 
 	m_cur_lba = new_lba;
 
 	if (diff == 0)
 		return TIME_BETWEEN_SECTORS;
 
-	attotime seek_time = (TIME_FULL_STROKE_SEEK * diff) / m_num_cylinders;
+	attotime seek_time = ((TIME_FULL_STROKE_SEEK - TIME_PER_ROTATION) * diff) / m_num_cylinders;
```

Comment:
Why are you subtracting the time per rotation here?  The `TIME_FULL_STROKE_SEEK` should be the time the mechanism takes to position the head for a seek of maximum distance.  It's not related to rotation time at all.  Now it's an oversimplification, because for small seeks the head positioning time is dominated by settling time, which is largely independent of the time it takes to seek.

But anyway, the time it takes for the desired sector to reach the head is applied below.  This calculation is supposed to involve just the head positioning time.

Reply:
That's weird... I could've SWORN I needed both of those changes to get MS-DOS's Setup working. o.o